### ISSUE TITLE
`URL.appendingPathExtension("")` appends a trailing dot

### ISSUE THREAD

**Issue by kateinoigakukun** [swiftlang/swift-foundation#1080]
```swift
import Foundation
let url = URL(fileURLWithPath: "/tmp/x")
print(url.appendingPathExtension(""))
// Swift 5.10: file:///tmp/x
// Swift 6.0: file:///tmp/x.
```

This behavior change might be tricky as `url.deletingPathExtension().appendingPathExtension(url.pathExtension)` won't be equal to `url`.

Https://github.com/swiftlang/swift-driver/pull/1756 revealed this change. Could you check if this is an intentional change? @jrflat

**Comment by jrflat** [comment]
Not intended, I'll push a fix, thanks for finding this!

**Comment by kateinoigakukun** [comment]
@jrflat Thanks! Can we cherry-pick this fix to 6.1 so that we will be able to remove a workaround once we drop 6.0 support?

**Comment by jrflat** [comment]
Yes sounds good I'll cherry-pick to 6.1
